examples/geostrophic_adjustment.jl

```diff
@@ -0,0 +1,132 @@
+# # Geostrophic adjustment using Oceananigans.HydrostaticFreeSurfaceMode l
+#
+# This example demonstrates how to simulate the one-dimensional geostrophic adjustment of a
+# free surface using `Oceananigans.HydrostaticFreeSurfaceModel`, which solves the hydrostatic
+# Boussinesq equations beneach a small-amplitude free surface, with boundary conditions
+# expanded around ``z = 0`` and free surface dynamics linearized under the assumption
+# ``η / H \ll 1``, where ``η`` is the free surface displacement, and ``H`` is the
+# total depth of the fluid.
+#
+# ## Install dependencies
+#
+# First let's make sure we have all required packages installed.
+
+# ```julia
+# using Pkg
+# pkg"add Oceananigans, JLD2, Plots"
+# ```
+
+# ## A one-dimensional domain
+#
+# We use a one-dimensional domain of geophysical proportions,
+
+using Oceananigans
+using Oceananigans.Utils: kilometers
+
+grid = RegularCartesianGrid(size = (128, 1, 1),
+                            x = (0, 1000kilometers), y = (0, 1000kilometers), z = (-400, 0),
+                            topology = (Bounded, Periodic, Bounded))
+
+# and Coriolis forces appropriate for mid-latitudes on Earth,
+
+coriolis = FPlane(f=1e-4)
+
+# ## Building a `HydrostaticFreeSurfaceModel`
+#
+# We use `grid` and `coriolis` to build a simple `HydrostaticFreeSurfaceModel`,
+
+using Oceananigans.Models: HydrostaticFreeSurfaceModel
+
+model = HydrostaticFreeSurfaceModel(grid=grid, coriolis=coriolis)
+#model = HydrostaticFreeSurfaceModel(grid=grid, coriolis=nothing)
+
+# ## A Geostrophc adjustment initial value problem
+#
+# We pose a geostrophic adjustment problem that consists of a Gaussian geostrophic
+# velocity and height field,
+
+Gaussian(x, L) = exp(-x^2 / 2L^2)
+
+U = 0.1 # geostrophic velocity
+L = grid.Lx / 40 # Gaussian width
+x₀ = grid.Lx / 4 # Gaussian center
+
+vᵍ(x, y, z) = - U * (x - x₀) / L * Gaussian(x - x₀, L)
+
+g = model.free_surface.gravitational_acceleration
+
+a = coriolis.f * U * L / g # free surface amplitude
```

Comment:
why is the surface elevation defined via this combination of parameters?
above we state that `η << Η`... are we sure that `a` defined as such is much less than `H`? I think it'll be clearer to just define `a = 1e-3 * grid.Lz` or something like that...

Reply:
perhaps also use `η₀` instead of `a`?

Reply:
Ah, this is the geostrophic amplitude; it's part of the steady geostrophic solution in which v and eta are in balance. Then I double eta so we see both a steady solution + waves. What do you suggest instead?

Reply:
oh OK I see! forget my remark then!